bundler/spec/install/gemfile_spec.rb

```diff
@@ -27,6 +27,20 @@
       ENV["BUNDLE_GEMFILE"] = "NotGemfile"
       expect(the_bundle).to include_gems "myrack 1.0.0"
     end
+
+    it "respects lock_file and BUNDLE_GEMFILE_LOCK" do
+      gemfile bundled_app("NotGemfile"), <<-G
+        lock_file "ReallyNotGemfile.lock"
+        source "https://gem.repo1"
+        gem 'myrack'
+      G
+
+      bundle :install, gemfile: bundled_app("NotGemfile")
+
+      ENV["BUNDLE_GEMFILE"] = "NotGemfile"
+      ENV["BUNDLE_GEMFILE_LOCK"] = "ReallyNotGemfile.lock"
+      expect(the_bundle).to include_gems "myrack 1.0.0"
+    end
```

Comment:
It seems that we need to split this to `lock_file` only case and `BUNGLE_GEMFILE_LOCK` case. If we have both of them and they use the same value, we can't test `lock_file` and `BUNDLE_GEMFILE_LOCK`.

How about splitting this to `lock_file` only case and `BUNDLE_GEMFILE_LOCK` only case`?

Reply:
We have a test for the `lock_file` only case in `bundler/spec/commands/install_spec.rb‎`. I'm not sure whether it would make sense to test the `BUNDLE_GEMFILE_LOCK` only case. If you aren't using `lock_file` in the `Gemfile`, `bundler` is going to append `.lock` to create the lock file path. I think it only makes sense to use `BUNDLE_GEMFILE_LOCK` if you are also using `lock_file` to change the default lock file path, unless you are moving the location of the lock file after it is created. We could have a test doesn't use `lock_file`,  and renames the lock files afterward, and then test that the renamed file works with `BUNDLE_GEMFILE_LOCK`. Is that what you'd like to see?

Reply:
Understood, thank you for explaining. I added a spec to test that `BUNDLE_GEMFILE_LOCK` correctly sets the lock file to create, if `lock_file` is not used in `Gemfile`.